v1/api/index.ts

```diff
@@ -0,0 +1,69 @@
+import { TypeBoxTypeProvider } from '@fastify/type-provider-typebox'
+import fastify, { FastifyBaseLogger, FastifyInstance, RawReplyDefaultExpression, RawRequestDefaultExpression, RawServerDefault } from 'fastify'
+import { siteRoutes } from './sites'
+import multipart from '@fastify/multipart'
+import swagger from '@fastify/swagger'
+import swagger_ui from '@fastify/swagger-ui'
+import { adminRoutes } from './admin'
+import { publisherRoutes } from './publisher'
+
+export type FastifyTypebox = FastifyInstance<
+RawServerDefault,
+RawRequestDefaultExpression<RawServerDefault>,
+RawReplyDefaultExpression<RawServerDefault>,
+FastifyBaseLogger,
+TypeBoxTypeProvider
+>
+
+interface APIConfig {
+  useLogging: boolean
+  useSwagger: boolean
+}
+
+async function apiBuilder ({ useLogging, useSwagger }: Partial<APIConfig>): Promise<FastifyTypebox> {
+  const server = fastify({ logger: useLogging }).withTypeProvider<TypeBoxTypeProvider>()
```

Comment:
Part of our milestone is to integrate analytics into our stack. Hypha typically uses prometheus with a graphana visualizer. Could you look into integrating something like https://gitlab.com/m03geek/fastify-metrics and seeing if we can pass something down to all the APIs so they can add more logs to the request.

This would be useful to track how long API calls take in general.

It might be useful to do the initialization of the prometheus client outside of the apiBuilder (if possible) so that we can pass it to the protocol backends too.

Reply:
We'll figure this out as part of another PR in https://github.com/hyphacoop/distributed-press-organizing/issues/60